benches/main.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE-BSD-3-Clause file.
+//
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+extern crate criterion;
+extern crate linux_loader;
+extern crate vm_memory;
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+use linux_loader::configurator::pvh::PvhBootConfigurator;
+use linux_loader::configurator::{BootConfigurator, BootParams};
+use linux_loader::loader::elf::start_info::{hvm_memmap_table_entry, hvm_start_info};
+use linux_loader::loader::elf::Elf;
+use linux_loader::loader::KernelLoader;
+use std::io::Cursor;
+use vm_memory::{Address, GuestAddress, GuestMemoryMmap};
+
+const E820_RAM: u32 = 1;
+const MEM_SIZE: usize = 0x100_0000;
+const XEN_HVM_START_MAGIC_VALUE: u32 = 0x336ec578;
+
+fn create_guest_memory() -> GuestMemoryMmap {
+    GuestMemoryMmap::from_ranges(&[(GuestAddress(0x0), MEM_SIZE)]).unwrap()
+}
+
+fn create_elf_pvh_image() -> Vec<u8> {
+    include_bytes!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/src/loader/x86_64/elf/test_elfnote.bin"
```

Comment:
Would it make sense to also test with a closer to real-life image? I was thinking we might be able to reuse the image that we are downloading in the buildkite hooks.

Also, I don't know if you were already planning this, but would it make sense to test different scenarios: loading bzimage, with/without pvh, others that might come up?

Reply:
Of course. I intend to benchmark all the publicly exposed functionalities, including the bzImage load. I only started with 2 tests to prove the approach on the infrastructure (need *something* to run)